<?xml version="1.0" ?>
<robot name="belaaaaaaaaaaaaaaaaaaaaaaazik">

  <link name="odom"/>

  <joint name="odom_to_base" type="fixed">
    <parent link="odom"/>
    <child link="base_link"/>
  </joint>

  <link name="base_link">
    <visual>
      <geometry>
        <box size="1.0 0.5 0.2"/>
      </geometry>
      <material name="blue">
        <color rgba="0.0 0.0 1.0 1.0"/>
      </material>
    </visual>
  </link>

  <link name="wheel_front_left">
    <visual>
      <geometry>
        <cylinder radius="0.3" length="0.1"/>
      </geometry>
      <material name="black">
        <color rgba="0.0 0.0 0.0 1.0"/>
      </material>
    </visual>
  </link>

  <link name="wheel_front_right">
    <visual>
      <geometry>
        <cylinder radius="0.3" length="0.1"/>
      </geometry>
      <material name="black">
        <color rgba="0.0 0.0 0.0 1.0"/>
      </material>
    </visual>
  </link>

  <link name="wheel_rear_left">
    <visual>
      <geometry>
        <cylinder radius="0.3" length="0.1"/>
      </geometry>
      <material name="black">
        <color rgba="0.0 0.0 0.0 1.0"/>
      </material>
    </visual>
  </link>

  <link name="wheel_rear_right">
    <visual>
      <geometry>
        <cylinder radius="0.3" length="0.1"/>
      </geometry>
      <material name="black">
        <color rgba="0.0 0.0 0.0 1.0"/>
      </material>
    </visual>
  </link>

  <joint name="rack_pinion_joint" type="revolute">
    <parent link="base_link"/>
    <child link="wheel_front_left"/>
    <origin xyz="0.4 0.25 0.1" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="10.0" lower="-1.57" upper="1.57" velocity="1.0"/>
  </joint>

  <joint name="joint_front_right" type="revolute">
    <parent link="base_link"/>
    <child link="wheel_front_right"/>
    <origin xyz="0.4 -0.25 0.1" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="10.0" lower="-1.57" upper="1.57" velocity="1.0"/>
  </joint>

  <joint name="wheel_rear_left" type="continuous">
    <parent link="base_link"/>
    <child link="wheel_rear_left"/>
    <origin xyz="-0.4 0.25 0.1"/>
    <axis xyz="0 0 1"/>
  </joint>

  <joint name="wheel_rear_right" type="continuous">
    <parent link="base_link"/>
    <child link="wheel_rear_right"/>
    <origin xyz="-0.4 -0.25 0.1"/>
    <axis xyz="0 0 1"/>
  </joint>

</robot>
